WordPress/gutenberg
# TextTransformControl/TextDecorationControl: Migrate to ToggleGroupControl

Part of #34345 #41973 
Closes #36735

⚠️ Will need to merge together with #43331

## What?

Replaces the internal implementation of TextTransformControl and TextDecorationControl to a ToggleGroupControl.

## Why?

The previous implementation allowed a "click to unselect" interaction, but failed to semantically capture that the options were mutually exclusive (#36735). We've decided to let go of the "click to unselect" interaction (in favor of the higher-level reset actions), so we can use a standard ToggleGroupControl. So this now becomes a proper `radio`.

By using a ToggleGroupControl, we'll also get the large size variants which will be required to upsize the Typography controls in the editor.

### Aside on TextDecorationControl

Technically, `text-decoration` values do not need to be mutually exclusive. But since the previous implementation behaves as mutually exclusive, and because it's highly unlikely that anyone would want an `underline line-through`, we'll be sticking to a `radio` at least for now.

## Testing Instructions

1. `npm run storybook:dev` and see the stories for TextTransformControl/TextDecorationControl.
2. `npm run dev` and add a Code block to the editor. Add the "Decoration" and "Letter case" controls to the Typography section in the Inspector. The controls should work as expected.

## Screenshots or screencast 

<img width="262" alt="New TextTransformControl/TextDecorationControl components" src="https://user-images.githubusercontent.com/555336/185141405-dc39dc4b-43d3-443c-9a9d-667a9840ddcd.png">
